x/dsmr/storage.go

```diff
@@ -32,12 +32,15 @@ type Verifier[T Tx] interface {
 	Verify(chunk Chunk[T]) error
 }
 
-var _ Verifier[Tx] = (*NoVerifier[Tx])(nil)
+var _ Verifier[Tx] = (*ChunkVerifier[Tx])(nil)
 
-type NoVerifier[T Tx] struct{}
+type ChunkVerifier[T Tx] struct {
+	networkID uint32
+	chainID   ids.ID
+}
 
-func (NoVerifier[T]) Verify(Chunk[T]) error {
-	return nil
+func (c ChunkVerifier[T]) Verify(chunk Chunk[T]) error {
+	return chunk.Verify(c.networkID, c.chainID)
```

Comment:
Should the verification logic be here so that the functionality is separated from the data?

Reply:
Disregard - it's currently just a syntactic check, so better to keep it on chunk itself and closer to the sign implementation